hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieArchivedTimeline.java

```diff
@@ -182,8 +183,11 @@ private String getMetadataKey(String action) {
           //read the avro blocks
           while (reader.hasNext()) {
             HoodieAvroDataBlock blk = (HoodieAvroDataBlock) reader.next();
-            // TODO If we can store additional metadata in datablock, we can skip parsing records
-            // (such as startTime, endTime of records in the block)
+            if (isDataOutOfRange(blk, filter)) {
```

Comment:
We only know that the data is out of range from the block and not from the whole file right ? So, this should be continue instead of break ?

Reply:
No. In the current implementation, the first block tracks range for entire fire. In some cases there are lot of archived files and its much faster to skip entire file when looking at older ranges. 

The overhead of storing metadata on every block seemed high. By default, we are grouping 10 records into one block. That translates to 10KB in size. Header on every block with min/max is adding 40 bytes overhead. So, 0.4% overhead seemed  high to me. Let me know if you think we can ignore overhead. I can move this to per block

Reply:
I added a comment above which I think addresses this as well

Reply:
addressed. added a unit test for this scenario